script.js

```diff
@@ -0,0 +1,228 @@
+// Basements (negatives) stay the same as the universal level.
+
+function getTheFloorInTheEuropeanSystem(americanFloorNumber) {
+    if (americanFloorNumber <= 0) {
+        return americanFloorNumber;
+    }
+    if (americanFloorNumber <= 12) {
+        return americanFloorNumber - 1;
+    }
+    if (americanFloorNumber >= 14) {
+        return americanFloorNumber - 2;
+    }
+}
+
+console.log(getTheFloorInTheEuropeanSystem(14));
+
+// 7. In the rock-paper-scissors.js file.
+
+// 8. Write the isValidPassword function.
+// Make sure that the password is at least 8 characters long and contains at least one number and one uppercase letter.
+
+function getLettersAmount(word) {
+    return word.length;
+}
+
+function getUppercaseLetter(word) {
+    const uppercaseLetters =  /[A-Z]/g;
+    const matchingResult = word.match(uppercaseLetters);
+    if (matchingResult === null) {
+        return 0;
+    }
+    return matchingResult.length;
+}
+
+function getNumber(word) {
+    const numbers = /\d/g;
+    const matchingResult = word.match(numbers);
+    if (matchingResult === null) {
+        return 0;
+    }
+    return matchingResult.length;
+}
+
+function isValidPassword(password) {
+    return getLettersAmount(password) >= 8 &&
+        getUppercaseLetter(password) >= 1 &&
+        getNumber(password) >= 1;
+}
+
+console.log(isValidPassword('Password123'));
+
+// 9. In the getPasswordStrength.js file.
+
+// 10. Create the getTotalOrderCost function that returns the total cost of the order and accepts the following arguments:
+// - the base cost of the items
+// - whether the customer is a VIP member - they get a 5% discount
+// - how many loyalty points the customer wants to use - each one of them reduces the cost by a cent (the final cost should not be lower than $0)
+// - is the package going to be shipped internationally - shipping internationally costs $10 (otherwise, it costs $5)
+// Make sure to apply the VIP discount before the loyalty points. The VIP discount does not apply to shipping.
+
+function isVipSubscriptionActive(vipSubscription) {
+    if (vipSubscription === 'active') {
+        return 0.05;
+    }
+    return 0;
+}
+
+function getLoyaltyPoints(pointAmount) {
+    if (pointAmount) {
+        return pointAmount;
+    }
+    return 0;
+}
+
+function isShippedInternationally(shippingType) {
+    if (shippingType === 'international') {
+        return 10;
+    }
+    return 5;
+}
+
+function getTotalOrderCost(baseCost, vipMember, loyaltyPoints, shipping) {
+    const fivePercentDiscount = isVipSubscriptionActive(vipMember) * baseCost;
+    const costReduceThroughLoyaltyPoints = getLoyaltyPoints(loyaltyPoints) / 100;
+    const shippingType = isShippedInternationally(shipping);
+    return baseCost - fivePercentDiscount - costReduceThroughLoyaltyPoints + shippingType;
```

Comment:
What if `costReduceThroughLoyaltyPoints` is bigger than the `baseCost`?

Reply:
The getTotalOrderCost function returns the negativ value. Okay, I have to change the code a bit .